shoes-swt/lib/shoes/swt/generate_backend.rb

```diff
@@ -1,7 +1,35 @@
 require 'rbconfig'
 
-def generate_backend(path)
-  options = "-J-XstartOnFirstThread" if RbConfig::CONFIG["host_os"] =~ /darwin/
+class Shoes
+  module SelectedBackend
+    class << self
+      def validate
+        if RUBY_ENGINE != "jruby"
+          puts <<-EOS
 
-  "jruby #{options} #{path}/shoes-swt"
+*******************************************************************************
+The shoes-swt backend requires a 9.X version of JRuby.
+
+You are running the following Ruby instead:
+
+  #{RUBY_DESCRIPTION}
+
+Please download JRuby from http://jruby.org (or your favorite Ruby version
+manager), reinstall the `shoes` gem on that Ruby, and try again.
+
+Sorry for the inconvenience!
```

Comment:
I like putting it right here, that's nice!

I think we might also link to our README. My experience with long error messages/warnings is that they get out of sync quickly and somewhat unwieldly while the README is more up to date and covers more ground... or should :)

To avoid misconceptions, I think this is concise enough to stay just thinking that a link to the README for further information might be helpful.